lib/spack/spack/solver/asp.py

```diff
@@ -2987,11 +2994,7 @@ class CompilerParser:
     def __init__(self, configuration) -> None:
         self.compilers: Set[KnownCompiler] = set()
         for c in all_compilers_in_config(configuration):
```

Comment:
@alalazo @scheibelp would this be the juncture where we would transition to using the new compiler interface? I.e. would we pull from the repo of installed spack packages and filter by compiler as well as composing package classes from the externals yaml? Or are we pulling compilers from the configs still? This is one of the several points where it's not clear how this new compiler interface is actually supposed to be integrated with the rest of Spack.